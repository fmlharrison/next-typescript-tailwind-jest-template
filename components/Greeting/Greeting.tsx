export const Greeting = ({ tools }: { tools: string[] }) => {
  const toolsAsSentence = new Intl.ListFormat().format(tools);
  return (
    <h1 className="font-sans text-3xl text-center" data-testid="greeting">
      Hello Next.js with {toolsAsSentence} ðŸ‘‹
    </h1>
  );
};
