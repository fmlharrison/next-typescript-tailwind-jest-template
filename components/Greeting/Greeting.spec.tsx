import { render, screen } from "@utils/test-utils";

import { Greeting } from "./Greeting";

describe("Greetings", () => {
  it("should render the heading", () => {
    render(<Greeting tools={["Jest", "React Testing Library"]} />);

    const heading = screen.getByTestId("greeting");

    expect(heading).toMatchSnapshot();
  });

  it("should render the heading", () => {
    render(<Greeting tools={["Jest", "React Testing Library"]} />);

    const heading = screen.getByTestId("greeting");

    expect(heading.textContent).toBe(
      "Hello Next.js with Jest and React Testing Library ðŸ‘‹"
    );
  });
});
