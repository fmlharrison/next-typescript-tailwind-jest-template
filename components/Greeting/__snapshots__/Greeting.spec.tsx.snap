// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greetings should render the heading 1`] = `
<h1
  class="font-sans text-3xl text-center"
  data-testid="greeting"
>
  Hello Next.js with 
  Jest and React Testing Library
   ðŸ‘‹
</h1>
`;
